src/main/java/com/bgsoftware/superiorskyblock/core/menu/button/impl/menu/BiomeButton.java

```diff
@@ -83,6 +88,19 @@ public void onButtonClick(SuperiorSkyblockPlugin plugin, MenuBiomes superiorMenu
             return;
         }
 
+        File file = new File(plugin.getDataFolder(), "menus/biomes.yml");
+        YamlConfiguration cfg = YamlConfiguration.loadConfiguration(file);
+
+        if(cfg.isConfigurationSection("biome-environment")) {
+            ConfigurationSection configurationSection = cfg.getConfigurationSection("biome-environment");
+            String environment = configurationSection.getString(event.getResult().name().toUpperCase(Locale.ENGLISH), "Normal");
+
+            if(clickedPlayer.getWorld() != null && clickedPlayer.getWorld().getEnvironment() != World.Environment.valueOf(environment.toUpperCase())) {
```

Comment:
Instead of calling String#toUpperCase everytime you interact with the environment string, you should once call on it String#toUpperCase and that's it. Also, The method should always have Locale.ENGLISH as a parameter.

Reply:
> Instead of calling String#toUpperCase everytime you interact with the environment string, you should once call on it String#toUpperCase and that's it. Also, The method should always have Locale.ENGLISH as a parameter.

Definitely yes. Conversion of diamond as DİAMOND will cause an invalid Material error, especially on a machine using a Turkish operating system.

Unfortunately, since there are many people who make this mistake, I have to use a few more parameters when starting my server.

`-Duser.language=en -Duser.country=US`